lib/spack/spack/installer.py

```diff
@@ -1715,9 +1810,24 @@ def build_process(pkg, kwargs):
 
                         # Redirect stdout and stderr to daemon pipe
                         phase = getattr(pkg, phase_attr)
-                        phase(pkg.spec, pkg.prefix)
 
-            echo = logger.echo
+                        # Catch any errors to report to logging
+                        try:
+                            phase(pkg.spec, pkg.prefix)
+                            if monitor:
+                                monitor.send_phase(pkg, phase_name, log_file, "SUCCESS")
```

Comment:
this and the sends of failures above could be callouts to some type of general hook instead of just something for monitor -- is that easy to do ?

Reply:
If I knew what a hook was, maybe? :P

Reply:
see above